vaktija-desklet@MuxBH28/files/vaktija-desklet@MuxBH28/desklet.js

```diff
@@ -0,0 +1,201 @@
+const Desklet = imports.ui.desklet;
+const St = imports.gi.St;
+const Gio = imports.gi.Gio;
+const ByteArray = imports.byteArray;
+
+function VaktijaDesklet(metadata, desklet_id) {
+    this._init(metadata, desklet_id);
+}
+
+VaktijaDesklet.prototype = {
+    __proto__: Desklet.Desklet.prototype,
+    lastFetchedData: null,
+    _init: function (metadata, desklet_id) {
+        global.log("Initializing Vaktija desklet");
+        Desklet.Desklet.prototype._init.call(this, metadata, desklet_id);
+
+        this.setupUI();
+        this.fetchData();
+        this.setupMidnightTimer();
+        this.setupTimeUpdateInterval();
+        this.setupCountdownUpdateInterval();
+    },
+
+    setupUI: function () {
+        global.log("Setting up UI");
+        this.window = new St.Bin();
+        this.text = new St.Label({ style_class: 'label' });
+        this.text.set_text("Fetching data...");
```

Comment:
Take a look at other applets for how to implement proper support for translatable strings. Ask if you have any questions.

Reply:
This is my first desklet and right now it works as expected but of course I am gonna be improving it.
Almost always I am using other desklets as a go to when fixing something.
If you have some more documentations (apart from cinnamon official), feel free to share with me please.

I will probably need your help with some things, especially with settings scheme which I could not figure out (usually [TypeError: this.settings.get_int is not a function])

Thank You

Reply:
> If you have some more documentations (apart from cinnamon official), feel free to share with me please.

At the moment, I would refer to the other desklets, especially if there is one that implements a feature you would like to implement yourself.



> I will probably need your help with some things, especially with settings scheme which I could not figure out (usually [TypeError: this.settings.get_int is not a function])

Feel free to ask and we can work through it together. I can link you to code that exists already that works that would hopefully get you unblocked.



> This is my first desklet

This is fine. Just remember that you don't need to rush.